### PR TITLE
feat(Cmake/Build): Use interface targets for inheriting flags and definitions

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
Co-authored-by: Naios <Naios@users.noreply.github.com>

Based TC commit https://github.com/TrinityCore/TrinityCore/commit/e60c1f84a7e13577a82046b02b214f9466d20a7c

##### CHANGES PROPOSED:
- Remove the usage of global flags and definitions
  set through the CMAKE_CXX_FLAGS variable.
- Use cmake target compile features for setting the
  CXX standard automatically.

##### ISSUES ADDRESSED:
- Closes

##### TESTS PERFORMED:
- [x] Build in win VS 2019

##### HOW TO TEST THE CHANGES:
- Build this

##### KNOWN ISSUES AND TODO LIST:
- [x] Test build in Linux

##### Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan please add @Naios as co-author

**Comment by Winfidonarleyan** [comment]
Yes yes, done

File: deps/bzip2/CMakeLists.txt [hunk 1]
```diff
@@ -33,6 +33,10 @@ else()
     PUBLIC
       ${CMAKE_CURRENT_SOURCE_DIR})
 
+  target_link_libraries(bzip2
+    PRIVATE
+      trinity-dependency-interface)
```
[the remark this answers is not in the document]
Comment (Winfidonarleyan) [reply deps/bzip2/CMakeLists.txt]: `acore-` in all

[the remark this answers is not in the document]
Comment (Viste) [reply deps/bzip2/CMakeLists.txt]: github is go insane, i write it last night

[the remark this answers is not in the document]
Comment (Winfidonarleyan) [reply deps/bzip2/CMakeLists.txt]: hah

**Comment by Winfidonarleyan** [comment]
@pklloveyou Can you build this?

**Comment by pklloveyou** [comment]
My Cmake changed a lot of content,, may not match

**Comment by pklloveyou** [comment]
I downloaded the latest one and the build went wrong

**Comment by Winfidonarleyan** [comment]
@pklloveyou Update to d62fb8ed865fc4daa36a1d18bedd41457f9720bc

**Comment by pklloveyou** [comment]
TIM截图20190907012135
The problem remains

**Comment by Winfidonarleyan** [comment]
Try delete Cache in Cmake and rebuild.
`trinity-*` not found in PR

**Comment by pklloveyou** [comment]
> Try delete Cache in Cmake and rebuild.
> `trinity-*` not found in PR

The problem remains

**Comment by pklloveyou** [comment]
TIM截图20190907014645

**Comment by Winfidonarleyan** [comment]
Hah, okay

**Comment by Winfidonarleyan** [comment]
```========== Сборка: успешно: 13, с ошибками: 0, без изменений: 8, пропущено: 2 ==========```
0 errors, 0 warnings.
Build with tools.
MS VS 2019 16.2.4

**Comment by pklloveyou** [comment]
okey~I forgot about it.
https://github.com/azerothcore/azerothcore-wotlk/pull/2255/commits/19e5618815d6c78493554779e9c085ad06f510a4

**Comment by pklloveyou** [comment]
Successful build, no problems found

**Comment by Winfidonarleyan** [comment]
Very good, thanks for testing

**Comment by BarbzYHOOL** [comment]
@pklloveyou are you on our discord? if yes, send me a message there !

**Comment by pklloveyou** [comment]
@BarbzYHOOL ?????

**Comment by FrancescoBorzi** [comment]
@pklloveyou do you use the "Discord" chat ?

https://discord.gg/gXNKCG5

**Comment by pklloveyou** [comment]
Can not be used in China :(

**Comment by pklloveyou** [comment]
I'm ready.

**Pull request merged**